core/templates/dev/head/pages/exploration_editor/editor_tab/SolutionVerificationService.js

```diff
@@ -29,8 +29,9 @@ oppia.factory('SolutionVerificationService', [
         var rulesServiceName = (
           angularNameService.getNameOfInteractionRulesService(interactionId));
         var rulesService = $injector.get(rulesServiceName);
-        AnswerClassificationService.getMatchingClassificationResult(
-          explorationId, state, correctAnswer, true, rulesService
+        return (
+          AnswerClassificationService.getMatchingClassificationResult(
+          explorationId, state, correctAnswer, true, rulesService)
```

Comment:
I'm slightly confused by the indentation and outer parens here. How about:

        return AnswerClassificationService.getMatchingClassificationResult(
          explorationId, state, correctAnswer, true, rulesService
        ).then(function(result) {
          ...

Edit: Actually I'm a bit confused by this change in general. Why are you returning this in the first place?